x-pack/plugins/security/server/authentication/http_authentication/http_authorization_header.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest } from '../../../../../../src/core/server';
+
+export class HTTPAuthorizationHeader {
+  constructor(public scheme: string, public credentials: string) {}
+
+  static parseFromRequest(request: KibanaRequest) {
+    const authorizationHeaderValue = request.headers.authorization;
+    if (!authorizationHeaderValue || typeof authorizationHeaderValue !== 'string') {
+      return null;
+    }
+
+    const [scheme] = authorizationHeaderValue.split(/\s+/);
+    const credentials = authorizationHeaderValue.substring(scheme.length + 1);
+
+    return new HTTPAuthorizationHeader(scheme, credentials);
```

Comment:
With the old implementation of `getHTTPAuthenticationScheme`, `.toLowerCase()` was being called when initially parsing the scheme from the request's headers. This felt inconsistent when I began to use the `HTTPAuthorizationHeader` class and `.toString()` to create the `Authorization` headers themselves. It has the effect of requiring consumers to call `.toLowerCase()`, or instead use `.localeCompare`, to treat this as being case insensitive. 

@azasypkin I'm interested in how you feel about this.

Reply:
It's fine by me as long as we mention this in the property JS docs. I'm curious if there is any reason we cannot do `toLowerCase()` for both `scheme` and `credentials` in `HTTPAuthorizationHeader` constructor though?

Reply:
We definitely can do this in the constructor. When we were previously building the `Authorization` header manually within the auth providers, we were using a capitalized version, for example: `Bearer ${accessToken}`. Changing this behavior for the sake of making matching the `scheme` easier felt wrong. However, I don't want to be introducing future bugs by being pedantic... I wish there was `CaseInsensitiveString` in JavaScript which overrode `equals`. etc. but AFAIK there is no such thing, or no ability to create one ourselves which would work with things like a `Map` and a `Set`.

Reply:
> Changing this behavior for the sake of making matching the scheme easier felt wrong. However, I don't want to be introducing future bugs by being pedantic... 

Yeah, I agree. We use and compare schemes just in a couple of places, I think the risk of accidentally breaking something in this area may be negligible.